sdk/core/ts-http-runtime/src/auth/credentials.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { OAuth2Flow } from "./authFlows.js";
+
+/**
+ * Options used when creating and sending get OAuth 2 requests for this operation.
+ */
+export interface GetOAuth2TokenOptions {
+  /** Abort signal for the request */
+  abortSignal?: AbortSignal;
+}
+
+/**
+ * Options used when creating and sending get bearer token requests for this operation.
+ */
+export interface GetBearerTokenOptions {
+  /** Abort signal for the request */
+  abortSignal?: AbortSignal;
+}
+
+/**
+ * Credential for OAuth2 authentication flows.
+ */
+export interface OAuth2TokenCredential<TFlows extends OAuth2Flow> {
+  /**
+   * Gets an OAuth2 token for the specified flows.
+   * @param flows - The OAuth2 flows to use.
+   * @param options - Options for the request.
+   * @returns - a valid access token which was obtained through one of the flows specified in `flows`.
+   */
+  getOAuth2Token(flows: TFlows[], options?: GetOAuth2TokenOptions): Promise<string>;
```

Comment:
why are flows generic here? Can't we just use the union type directly?

```suggestion
  getOAuth2Token(flows: OAuth2Flow[], options?: GetOAuth2TokenOptions): Promise<string>;
```